models/users.js

```diff
@@ -96,27 +96,32 @@ module.exports = {
   },
 
   // Helper function for /api/friendships to add friends to friendships db
-  addFriend: function (user1, user2) {
+  addFriend: async function (user1, user2) {
     const addFriendship = {
-      text: 'INSERT INTO friendships(user1, user2) VALUES ($1, $2) ',
+      text: 'INSERT INTO friendships(user1, user2) VALUES ($1, $2)',
       values: [user1, user2]
     }
     return db.query(addFriendship)
   },
 
-  // delete user account
-  deleteUser: function (userSecret) {
-
-  },
-
-  // remove all of a users friendships
-  removeAllFriendships: function (userSecret) {
-
+  // delete user account and all existing friendships
+  deleteUserAndFriendships: async function (userSecret) {
+		const deleteUserAndFriendships = {
+			text: 'DELETE FROM users USING users as u ' +
+			'LEFT JOIN friendships AS f ON  f.user1 = u.id ' +
+			'LEFT JOIN users as u1 ON f.user2 = u.id WHERE users.secret = $1',
+			values: [userSecret]
```

Comment:
Friendships should be automatically deleted because of the foreign key constraint "DELETE CASCADE". If they aren't deleted automatically, then we should fix the foreign key constraint in Friendships. 

It's possible that your local DB doesn't match the production DB settings. We should create a script that initializes the DB, so that our DB structure is encoded in github, and to make sure that our local and production DB environments are as similar as possible.

Reply:
Ahh your right. Ty for clarification.